### ISSUE TITLE
Question: Slow Catalog and Product pages when logged in 

### ISSUE THREAD

**Issue by chocolata** [OFFLINE-GmbH/oc-mall-plugin#1151]
Hi guys,

I've got some clients that seem to mention that whenever a user is logged in, the Catalog pages and Product pages load a lot slower.

I can reproduce this, but at the moment I'm clueless as to why this would happen.

Did anyone come across this yet?

Thanks.

**Comment by chocolata** [comment]
Hi guys,

Just following up on this. Does anyone else have issues with this?

**Comment by tobias-kuendig** [comment]
I'd start by looking at the query log and see if there are any slow queries.

**Comment by chocolata** [comment]
Hi @tobias-kuendig 

Thanks a lot. We've set up a slow query log and found the following entries across multiple sites:

```
# Time: 240820 10:06:31
# User@Host: xxxxx_puser[xxxxx_puser] @ localhost []
# Thread_id: 141747  Schema: xxxxx_pdb  QC_hit: No
# Query_time: 4.445662  Lock_time: 0.000043  Rows_sent: 1  Rows_examined: 1292674
# Rows_affected: 0  Bytes_sent: 0
SET timestamp=1724141191;
select * from `offline_mall_carts` where (`customer_id` = 1) and `offline_mall_carts`.`deleted_at` is null order by `created_at` desc limit 1;
```

I'm still investigating if this query originates from a custom plugin or not and will let you know if I find anything.

Do you recognize this query maybe?

**Comment by tobias-kuendig** [comment]
It looks like this is related to https://github.com/OFFLINE-GmbH/oc-mall-plugin/issues/280

You just have a lot of empty carts in your table.

**Comment by chocolata** [comment]
Hi @tobias-kuendig 

Thanks for clarifying. Looks like that is the case indeed! I didn't know carts are created just from a visit...

It's fairly common for sites to get these kind of spikes, so that might populate the cart table fairly quickly.
![image](https://github.com/user-attachments/assets/4d65959b-99be-47ff-94f3-a944d41e80fa)

I'll look at integrating your suggested solution into a snippet that deletes empty carts on a schedule.

```
$q = \OFFLINE\Mall\Models\Cart
            ->where('customer_id', null)
            ->where('updated_at', '<', now()->subDays(3))
            ->get()
            ;
```
But it wouldn't solve issues when there are spikes in traffic, because the empty carts would only be deleted after the fact, and in the three days after such a spike, the site would run slowly. 

We even saw payments failing because of time-outs.

Can we please consider a structural solution for this?

Thank you for all of your work.

**Comment by chocolata** [comment]
Hi, 

Just wanted to share what I'm using now. In the Plugin.php file of my custom plugin, I've set up this schedule to run every day.
At the moment it seems to work. Do you see anything I've missed as I'm only deleting entries from the `offline_mall_carts` table and do not take into account any relations...

```
    public function registerSchedule($schedule)
    {
        $schedule->call(function () {
            \OFFLINE\Mall\Models\Cart::where('customer_id', null)
                ->where('updated_at', '<', now()->subDays(3))
                ->delete();
        })->daily();
    }
```

**Comment by tobias-kuendig** [comment]
Thanks for bringing the issue up again, @chocolata. I was able to implement a simple fix that no longer creates carts for simple page views.

This was long overdue!

https://github.com/OFFLINE-GmbH/oc-mall-plugin/commit/107ed12130441e535af94b5843785340c2668a4f

Let me know if you find any issues with it.

**Comment by chocolata** [comment]
Thanks for your great work as always. Your fix seems to work fine. I truncated the `offline_mall_carts` table on my staging environment and ran some tests. A cart seems to be created upon adding an item to the cart, and so long as you are not logged in, the `session_id` field of the row is populated. As soon as you log in, that` session_id` is cleared, which seems logical. 

One thing I notice, that isn't really clear, is that some rows get their `deleted_at` value populated and others don't... I ran through the checkout three times, and the last time, the record wasn't marked as deleted. But this doesn't seem to be an issue.

But everything seems to be functional at the moment! The most important thing is that the `offline_mall_carts` table doesn't get flooded, and this solves that.

![image](https://github.com/user-attachments/assets/406f7ccb-ac93-48e2-9fbb-cd951d26c6d4)
